digital_land/commands.py

```diff
@@ -531,6 +533,232 @@ def collection_add_source(entry, collection, endpoint_url, collection_dir):
     add_source_endpoint(entry, directory=collection_dir)
 
 
+def is_url_valid(url, url_type):
+    if not url or url.strip() == "":
+        return False, f"The {url_type} must be populated"
+
+    parsed_url = urlparse(url)
+    # is  url scheme valid i.e start with http:// or https://
+    if parsed_url.scheme not in ["http", "https"] or not parsed_url.scheme:
+        return False, f"The {url_type} must start with 'http://' or 'https://'"
+
+    # does url have domain
+    if not parsed_url.netloc:
+        return False, f"The {url_type} must have a domain"
+    return True, None
```

Comment:
Think we might want to check to see if there is a top-level domain such as `.com` or `.gov.uk`. A url like `https://plymouth` seems to still pass instead of raising an error. Having looked into urlparse's `.netloc`, considers this a valid netloc, but it's still not a valid domain. Maybe we could have a check to see if the netloc contains a `.`. This check is probably not fool proof though

```suggestion
    parsed_url = urlparse(url)
    # is  url scheme valid i.e start with http:// or https://
    if parsed_url.scheme not in ["http", "https"] or not parsed_url.scheme:
        return False, f"The {url_type} must start with 'http://' or 'https://'"

    # does url have domain
    if not parsed_url.netloc:
        return False, f"The {url_type} must have a domain"
        
    if "." not in parsed_url.netloc:
        return False, f"The {url_type} must have a valid domain with a top-level domain (e.g., '.com')"

    return True, None
```